### PR TITLE
fixed bug where graph variables was being set to null on union operation

### PR DESCRIPTION
**Pull request by patofp**
This PR fixed a bug where the graph variable column was being set to null when performing a union operation.

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

### CODE REVIEW

File: modules/engine/src/main/scala/com/gsk/kg/engine/relational/relational.scala [hunk 1]
```diff
@@ -54,6 +54,7 @@ import simulacrum.typeclass
   def map[U: Encoder](df: A, fn: Row => U): A
   def flatMap[U: Encoder](df: A, fn: Row => TraversableOnce[U]): A
   def collect(df: A): Array[Row]
+  def show(df: A, truncate: Boolean): Unit
```
Comment (shihch) [inline modules/engine/src/main/scala/com/gsk/kg/engine/relational/relational.scala:57]: It's fine here but seems not related to the issue.

**Review by shihch — APPROVED** [review]
Looks good as far as I can tell.

File: modules/engine/src/main/scala/com/gsk/kg/engine/relational/relational.scala (hunk 1 above)
Comment (patofp) [reply modules/engine/src/main/scala/com/gsk/kg/engine/relational/relational.scala:57]: Yes, that is true. It doesn't have anything to do with the bug but is a useful method for debugging.

Strictly it shouldn't be on this PR as it is an enhancement and it has nothing to do with the bug-fix, but I think it doesn't do any harm.

What is your opinion, should we remove it from this PR?

Comment (shihch) [reply modules/engine/src/main/scala/com/gsk/kg/engine/relational/relational.scala:57]: Please go ahead to merge. I just ask for clarification. As long as the function is used, it does not matter with which pull request.

Comment (patofp) [reply modules/engine/src/main/scala/com/gsk/kg/engine/relational/relational.scala:57]: Cool!
